src/main/java/com/socialmedia/util/friendship/Graph.java

```diff
@@ -0,0 +1,84 @@
+package com.socialmedia.util.friendship;
+
+import com.socialmedia.model.ApplicationUser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.Stack;
+import java.util.Queue;
+import java.util.LinkedList;
+
+class Graph {
+  private Map<Vertex, List<Vertex>> adjVertices = new HashMap<>();
```

Comment:
If we have this Graph specifically for user, wouldn't it be better to have Map<String, List<Vertex>>  where String is username? We'll avoid a lot of unnecessary Vertex creations and will have more productive search across hashMap, won't we?